scheduler/src/cook/rest/api.clj

```diff
@@ -1420,9 +1429,13 @@
 
 (def leader-hostname-regex #"^([^#]*)#([0-9]*)#([a-z]*)#.*")
 
+(defn leader-selector->leader-id
+  [leader-selector]
+  (-> leader-selector .getLeader .getId))
```

Comment:
Is there always guaranteed to be a known leader? If not, should we generate a `503` error?

Reply:
We looked at the code and `.getLeader` returns a _dummy participant_ if there's no leader available; however, that dummy participant will return false on `.isLeader`, so we can use that to detect this scenario.

Reply:
Refactored to throw `IllegalStateException`, which is caught in the handler and results in a 503.